submission/persistence-filesystem/src/main/kotlin/ac/uk/ebi/biostd/persistence/filesystem/fire/FireFilesService.kt

```diff
@@ -1,111 +1,28 @@
-import ac.uk.ebi.biostd.persistence.filesystem.extensions.getOrPersist
 import ac.uk.ebi.biostd.persistence.filesystem.request.FilePersistenceRequest
 import ac.uk.ebi.biostd.persistence.filesystem.service.FileProcessingService
-import ebi.ac.uk.extended.model.ExtFile
-import ebi.ac.uk.extended.model.ExtFileType.DIR
-import ebi.ac.uk.extended.model.ExtFileType.FILE
-import ebi.ac.uk.extended.model.FireFile
-import ebi.ac.uk.extended.model.NfsFile
 import ebi.ac.uk.extended.model.ExtSubmission
-import ebi.ac.uk.io.ext.md5
-import ebi.ac.uk.io.ext.size
 import mu.KotlinLogging
-import org.zeroturnaround.zip.ZipUtil
-import uk.ac.ebi.fire.client.integration.web.FireWebClient
-import java.io.File
-import java.nio.file.Path
 
 private val logger = KotlinLogging.logger {}
 
 class FireFilesService(
-    private val fireTempDirPath: Path,
-    private val fireWebClient: FireWebClient,
+    private val fireService: FireService,
     private val fileProcessingService: FileProcessingService
 ) : FilesService {
     override fun persistSubmissionFiles(request: FilePersistenceRequest): ExtSubmission {
-        val (sub, _) = request
-        logger.info { "${sub.accNo} ${sub.owner} Persisting files of submission ${sub.accNo} on FIRE" }
+        val sub = request.submission
 
-        cleanSubmissionFolder(sub.accNo)
-        val config = FireFileProcessingConfig(
-            sub.accNo, sub.version, sub.owner, sub.relPath, fireTempDirPath, fireWebClient
-        )
-        val processed = fileProcessingService.processFiles(sub) { config.processFile(request.submission, it) }
+        logger.info { "${sub.accNo} ${sub.owner} Cleaning Ftp Folder for ${sub.accNo}" }
+        fireService.cleanFtp(sub)
+        logger.info { "${sub.accNo} ${sub.owner} Finished Ftp Folder for ${sub.accNo}" }
 
+        logger.info { "${sub.accNo} ${sub.owner} Persisting files of submission ${sub.accNo} on FIRE" }
+        val processed = fileProcessingService.processFiles(sub) { fireService.getOrPersist(request.submission, it) }
         logger.info { "${sub.accNo} ${sub.owner} Finished persisting files of submission ${sub.accNo} on FIRE" }
 
         return processed
     }
-
-    private fun cleanSubmissionFolder(accNo: String) {
-        fireWebClient
-            .findByAccNo(accNo)
-            .forEach { fireWebClient.unsetPath(it.fireOid) }
-    }
-}
-
-data class FireFileProcessingConfig(
-    val accNo: String,
-    val version: Int,
-    val owner: String,
-    val subRelPath: String,
-    val fireTempDirPath: Path,
-    val fireWebClient: FireWebClient
-)
-
-fun FireFileProcessingConfig.processFile(sub: ExtSubmission, file: ExtFile): ExtFile = when (file) {
-    is NfsFile -> processNfsFile(sub.relPath, file)
-    is FireFile -> reuseFireFile(file, sub.relPath)
-}
-
-fun FireFileProcessingConfig.processNfsFile(relPath: String, nfsFile: NfsFile): ExtFile {
-    logger.info { "$accNo $owner Persisting file ${nfsFile.fileName} with size ${nfsFile.file.size()} on FIRE" }
```

Comment:
We're missing these logs in the new class which I think is very useful for keeping track of the submission process